core/execution/market.go

```diff
@@ -1731,7 +1732,27 @@ func (m *Market) confirmMTM(
 				m.log.Error("Failed to perform bond slashing",
 					logging.Error(err))
 			}
-			m.broker.Send(events.NewLedgerMovements(ctx, transfers))
+			// if bond slashing occurred then amounts in "closed" will not be accurate
+			if len(transfers) > 0 {
+				m.broker.Send(events.NewLedgerMovements(ctx, transfers))
+				asset, _ := m.mkt.GetAsset()
+				closedRecalculated := make([]events.Margin, 0, len(closed))
```

Comment:
That might be what you mean about the events being used for internal stuff it's fine here, it's not very pretty, and it should probably defined somewhere else but these are just interface of data being carried around the core.

Reply:
Cool, thanks for confirming